examples/base-next-app-router-tailwind-ts/src/app/components.tsx

```diff
@@ -0,0 +1,157 @@
+'use client';
+import * as React from 'react';
+import clsx from 'clsx';
+import BaseSelect, { SelectProps, SelectOwnerState, SelectRootSlotProps } from '@mui/base/Select';
+import Option, { OptionProps, OptionOwnerState } from '@mui/base/Option';
+import BaseSlider, { SliderProps, SliderOwnerState } from '@mui/base/Slider';
+
+const SelectButton = React.forwardRef(function SelectButton<
+  TValue extends {},
+  Multiple extends boolean,
+>(props: SelectRootSlotProps<TValue, Multiple>, ref: React.ForwardedRef<HTMLButtonElement>) {
+  const { ownerState, ...other } = props;
+  return (
+    <button type="button" {...other} ref={ref}>
+      {other.children}
+      <svg
+        xmlns="http://www.w3.org/2000/svg"
+        width="24"
+        height="24"
+        viewBox="0 0 24 24"
+        fill="none"
+        className="text-base absolute h-full top-0 right-1"
+      >
+        <path d="m7 10 5 5 5-5H7z" />
+      </svg>
+    </button>
+  );
+});
+
+// the components exported from this file have function props which are non-serializable
+// therefore they are additionally wrapped with 'use client' here
+// https://nextjs.org/docs/getting-started/react-essentials#passing-props-from-server-to-client-components-serialization
+
+export function SelectOption({
+  slotProps = {
```

Comment:
IMO it's more readable to destructure props in the function body. It took me a while to figure out where does the function actually start.